cartographer_grpc/framework/server.cc

```diff
@@ -52,9 +61,10 @@ void Server::AddService(
     const std::string& service_name,
     const std::map<std::string, RpcHandlerInfo>& rpc_handler_infos) {
   // Instantiate and register service.
-  const auto result =
-      services_.emplace(std::piecewise_construct, std::make_tuple(service_name),
-                        std::make_tuple(service_name, rpc_handler_infos));
+  const auto result = services_.emplace(
+      std::piecewise_construct, std::make_tuple(service_name),
+      std::make_tuple(service_name, rpc_handler_infos,
+                      [this]() { return SelectNextEventQueueRoundRobin(); }));
```

Comment:
Is this modification of  current_event_queue_id_  thread-safe and should it be?

Reply:
Good catch! Fixed.